doc/source/handling-dependencies.rst

```diff
@@ -0,0 +1,216 @@
+
+.. code-block:: python
+
+  import ray
+
+  # /path/to/files is a directory on the local machine.
+  # /path/to/files/hello.txt contains the string "Hello World!"
+  ray.init(runtime_env={"working_dir": "/path/to/files"})
+
+  @ray.remote
+  def f():
+      return open("hello.txt").read()
+
+  print(ray.get(f.remote())) # Hello World!
+
+The example above starts a single-node Ray cluster on your local machine, but by specifying an address (e.g. ``ray.init("ray://123.456.7.89:10001", runtime_env=...)`` to connect to a remote cluster using Ray Client, the same code will still work, and ``f`` will be running on the remote cluster.
+
+The specified local directory will automatically be pushed to the cluster nodes when `ray.init()` is called.
+
+Workflow: Ray Library development
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Suppose you are developing a library ``my_module`` on Ray.
+A typical iteration cycle will involve making some changes to the source code of ``my_module`` and then running a Ray script to test the changes.  Perhaps your test requires a remote cluster.  To ensure your local changes show up on the cluster, you can use the ``py_modules`` field of runtime environments.
+
+.. code-block:: python
+
+  import ray
+  import my_module
+
+  ray.init("ray://123.456.7.89:10001", runtime_env={"py_modules": [my_module]})
+
+  @ray.remote
+  def test_my_module():
+      # No need to import my_module inside this function.
+      my_module.test()
+
+  ray.get(f.remote())
+
+.. _runtime-environments-api-ref:
+
+API Reference
+^^^^^^^^^^^^^
+
+The ``runtime_env`` is a Python dictionary including one or more of the following keys:
+
+- ``working_dir`` (str): Specifies the working directory for the Ray workers. This must either be an existing directory on the local machine with total size at most 100 MiB, or a path to a zip file stored in Amazon S3.
+  If a local directory is specified, it will be uploaded to each node on the cluster.
+  Ray workers will be started in their node's copy of this directory.
+
+  - Examples
+
+    - ``"."  # cwd``
+
+    - ``"/src/my_project"``
+
+    - ``"s3://path/to/my_dir.zip"``
+
+  Note: Setting a local directory per-task or per-actor is currently unsupported.
+
+  Note: If your local directory contains a ``.gitignore`` file, the files and paths specified therein will not be uploaded to the cluster.
+
+- ``py_modules`` (List[str|module]): Specifies Python modules to import in the Ray workers.  (For more ways to specify packages, see also the ``pip`` and ``conda`` fields below.)
+  Each entry must be either (1) a path to a local directory, (2) a path to a zip file stored in Amazon S3, or (3) a Python module object.  
+
+  - Examples
+
+    - ``"."``
+
+    - ``"/src/my_module"``
+
+    - ``"s3://path/to/my_module.zip"``
+
+    - ``my_module # Assumes my_module has already been imported, e.g. via 'import my_module'``
+
+  Note: Note: Setting options (1) and (3) per-task or per-actor is currently unsupported.
```

Comment:
This is confusing for me what are supported by job level / task and actor level. Maybe having a separate section to explain what APIs are supported? Like;

Supported APIs:

Jobs
- working dir
- conda env
- pymodule...

Per tasks/actors
- conda env